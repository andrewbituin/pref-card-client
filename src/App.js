import React, { Component } from "react";
import { Route } from "react-router";
import Home from "./components/Home";
import Login from "./components/Login";
import Register from "./components/Register";
import CardById from "./components/CardById";
import CardsList from "./components/CardsList";
import AddCardForm from "./components/AddCardForm";
import { CardsContextProvider } from "./context/CardsContext";
import PrivateRoute from './components/Utils/PrivateRoute';
import EditForm from "./components/EditForm";

class App extends Component {
  render() {
    return (
      <CardsContextProvider>
        <div>
          <Route exact path="/" render={props => <Home {...props}/>} />
          <Route
            exact
            path="/register"
            render={props => <Register {...props} />}
          />
          <PrivateRoute
            exact
            path="/card/:id"
            component={CardById}
          />
          <PrivateRoute
            exact
            path="/all"
            component={CardsList} 
            />
          <PrivateRoute
            exact
            path="/create-card"
            component={AddCardForm}
          />
          <PrivateRoute
            exact
            path="/edit/:id"
            component={EditForm}
          />
        </div>
      </CardsContextProvider>
    );
  }
}

export default App;
