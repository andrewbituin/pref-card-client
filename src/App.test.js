import React from "react";
import ReactDOM from "react-dom";
import App from "./App";
import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";

describe("App component", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <MemoryRouter>
        <App />
      </MemoryRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });
  it("renders the UI as expected", () => {
    const tree = renderer.create(
      <MemoryRouter>
        <App />
      </MemoryRouter>
    );
    expect(tree).toMatchSnapshot();
  });
});
