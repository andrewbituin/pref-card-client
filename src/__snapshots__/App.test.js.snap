// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders the UI as expected 1`] = `
<div>
  <nav
    className="navigation"
  >
    <div
      className="form-container"
    >
      <form
        className="login-form"
        onSubmit={[Function]}
      >
        <div
          role="alert"
        />
        <div
          className="username-container"
        >
          <label
            className="username-label"
          >
            Username 
          </label>
          <input
            className="userName"
            onChange={[Function]}
            type="text"
          />
        </div>
        <div
          className="password-container"
        >
          <label
            className="password-label"
          >
            Password 
          </label>
          <input
            className="password"
            onChange={[Function]}
            type="password"
          />
        </div>
        <div
          className="home-buttons"
        >
          <button
            className="login-submit-button"
            type="submit"
          >
            Log in
          </button>
          <a
            href="/register"
            onClick={[Function]}
          >
            <button
              className="register-button"
              type="click"
            >
              Register
            </button>
          </a>
        </div>
      </form>
    </div>
  </nav>
  <main>
    <header
      className="home-header"
    >
      <h1
        className="home-title"
      >
        prefcards
      </h1>
      <h2
        className="home-subtitle"
      >
        Reliable. Accessible. Accurate.
      </h2>
    </header>
    <section
      className="about-section "
    >
      <div
        className="column"
      >
        <h3>
          Experience simplicity and usablity.
        </h3>
        <p>
          Prefcards provides an easy way to generate, update and view surgical preference cards.
        </p>
      </div>
      <div
        className="column"
      >
        <h3>
          Work with team focused accessibility.
        </h3>
        <p>
          Prefcards allows you limit the users who are able to edit your preference card, ensuring consistency across procedures and allowing staff to come and go knowing that the surgeon's preferences for equipment and patient information is as it should be.
        </p>
      </div>
      <div
        className="column"
      >
        <h3>
          Log in and get started.
        </h3>
        <p>
          Create an account to get started.
        </p>
      </div>
    </section>
  </main>
</div>
`;
